### ISSUE TITLE
Expand button in Table and DataList components

### ISSUE THREAD

**Issue by jgiardino** [patternfly/patternfly-react#1163]
The expand buttons in the Table and DataList are similar patterns, and should have similar semantics associated with them.

The following aria attributes for the expand button should be used in both of these components:
- **`aria-controls="[id of controlled element]"`** -  This should be used when the controlled contents are not adjacent to the toggle that controls them, as in the Table and DataList. In some screen readers, the user is provided with a key command to shift focus to the controlled element. While the documentation allows multiple id values, there has been some questioning of this, since the benefit is in shifting focus to a specific element. Léonie Watson describes how this attribute should be used in an [article](https://tink.uk/using-the-aria-controls-attribute/), and also commented on the benefits and only using a single id in the second comment to this [article](http://www.heydonworks.com/article/aria-controls-is-poop).
  - This should be applied to both components. **UPDATE** This will be deferred for the Table component
- **`aria-label="Details"`** - This provides an accessible name for the button
  - This is currently defined in the Table component, but needs to be updated in the DataList component
- **`aria-labelledby="[id of row header] [id of expand button]"`** - This combines the row header / list item heading text with the button label to provide an accessible name that is context specific (e.g. "Node 1 Details" versus just "Details")
  - This is currently defined in the Table component, but the order of `id`s specified in the DataList component needs to be updated. 
- **Allow for translated strings** - The consumer should also be able to provide a translated string in this case. 
  - This could be split out as a separate issue if needed.
  - If the consumer provides a translatable string—e.g. `"{itemHeading} Details"`—then this should be defined as the value for `aria-label` and `aria-labelledby` MUST be removed.

**Comment by jgiardino** [comment]
There was some discussion of `aria-controls` in #821. I'm pasting comments from that here:

> With the aria-controls, that's not gonna be easy. Parents do not know about which children they open (To allow asynchronous loading of collapsed children). And what if there are multiple items collapsed on click <button aria-label="Details" aria-expanded="true" aria-controls=[id1 id2 id3] class="pf-c-button pf-m-plain pf-m-expanded" type="button">? However I have added id to expanded row content.

**Comment by rebeccaalpert** [comment]
@jgiardino - It looks like the DataList button lets you put in whatever you want for the aria-label. Should we take that control away from users or do you want "Details" as the default?

**Comment by jgiardino** [comment]
I was wondering if it's possible to have a default implementation that works, but to allow consumers to provide a custom string.

So by default, the button would be this, and announced as "Node 1 Details":
```
<button 
  id="ex-toggle1" 
  aria-label="Details" 
  aria-labelledby="ex-item1 ex-toggle1" 
  aria-expanded="true" 
  class="pf-c-button pf-m-plain" 
  type="button"
>
```
But if the consumer specifies a string for this button, then it should be this, and hopefully the string would include some reference to the main heading of that item so that the labels are unique:
```
<button 
  aria-label={consumer_string} 
  aria-expanded="true" 
  class="pf-c-button pf-m-plain" 
  type="button"
>
```

My goal is to not require consumers to think about how this button needs to be labeled and still have accessible labels, but also allow the consumer to provide a label (e.g. for translation). 

It would be worth getting feedback from some of the other react devs on how to do this, since I know this has been a topic of discussion, and I'm not sure what the current line of thinking is.

**Comment by jgiardino** [comment]
After chatting with @rebeccaalpert, this issue can be closed without applying `aria-controls="[id of controlled element]"` to the Table component, given that there's no easy way to implement that. This might be something worth considering in the future if enhancements are ever made that make this possible for either us to build in or enable the consumer to implement.
